cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -268,8 +269,13 @@ func (g *gitHubRepository) getLatestRelease() (string, error) {
 		}
 	}
 
+	// Fall back to looking for latest prereleases if no release has been cut or bail if we're already there
 	if latestTag == "" {
-		return "", errors.New("failed to find releases tagged with a valid semantic version number")
+		if g.acceptPrerelease {
```

Comment:
Wondering if there is a way to avoid the `acceptPrerelease` flag at repository level, because it might have side effects on subsequent calls on getLatestRelease.

What about computing both  `latestTag` and `latestPreReleaseTag` (new) during the for loop and return the latter if latestTag is null? 
Computing both values in a single call could avoid the nested call to `getLatestRelease` and thus the need of the `acceptPrerelease` flag ...

Reply:
yeah, that's probably fair. let me revisit this AM.

Reply:
@fabriziopandini yep, this is quite a bit cleaner. updated in amended commit.